gallery/tutorials/aspire_introduction.py

```diff
@@ -0,0 +1,675 @@
+# ``Volumes`` ``.map`` and ``.mrc`` are currently supported.  For
+# ``.npy`` Numpy can be used.
+
+# A low res example file is included in the repo as a sanity check.
+# We can instantiate this as an ASPIRE Volume instance using
+# ``Volume.load()``.
+file_path = os.path.join(os.getcwd(), "data", "clean70SRibosome_vol_65p.mrc")
+v = Volume.load(file_path)
+
+# %%
+# More interesting data requires downloading locally.  A common
+# starting dataset can be downloaded from EMDB at
+# `<https://www.ebi.ac.uk/pdbe/entry/emdb/EMD-2660>`_.  After
+# downloading the associated `map` file, unzip in local directory.  To
+# simplify things, this notebook defaults to a small low resolution
+# sample file instead.  Unfortunately real data can be quite large so
+# we do not ship it with the repo.
+
+# v = Volume.load("path/to/EMD-2660/map/emd_2660.map")
+
+# %%
+# Downsample Volume
+# ^^^^^^^^^^^^^^^^^
+# Here we downsample the above volume to a desired image size (32
+# should be good).
+
+img_size = 32
+
+# Volume.downsample() returns a new Volume instance.
+#   We will use this lower resolution volume later, calling it `v2`.
+v2 = v.downsample(img_size)
+# L is often used as short hand for image and volume sizes (in pixels/voxels).
+L = v2.resolution
+
+# %%
+# Plot Data
+# """""""""
+# For quick sanity checking purposes we can view some plots.
+#   We'll use three orthographic projections, one per axis.
+orthographic_projections = np.empty((3, L, L), dtype=v2.dtype)
+for i in range(3):
+    orthographic_projections[i] = np.sum(v2, axis=(0, i + 1))
+Image(orthographic_projections).show()
+
+# %%
+# ``Rotation`` Class
+# ------------------
+# While you may bring your own 3x3 matrices or generate manually (say
+# from your own Euler angles), ASPIRE has a `Rotation class
+# <https://computationalcryoem.github.io/ASPIRE-Python/aspire.utils.html#module-aspire.utils.rotation>`_
+# which can do this random rotation generation for us.  It also has
+# some other utility methods, including support for Rodrigues
+# rotations (ie, axis-angle).  Other ASPIRE components dealing with 3D
+# rotations will generally expect instances of ``Rotation``.
+#
+# A common task in Computational CryoEM is generating random
+# projections, which can be achieved by applying random 3D
+# rotations. The following code will generate some random rotations,
+# and use the ``Volume.project()`` method to return an ``Image``
+# instance representing the stack of projections.  We can display
+# projection images using the ``Image.show()`` method.
+
+from aspire.utils import Rotation
+
+num_rotations = 2
+rots = Rotation.generate_random_rotations(n=num_rotations, seed=12345)
+
+# %%
+# We can access the Numpy array holding the actual stack of 3x3 matrices:
+logging.info(rots)
+logging.info(rots.matrices)
+
+# %%
+# Using the zero-th (and in this case, only) volume, compute
+# projections using the stack of rotations:
+projections = v.project(0, rots)
```

Comment:
Reminds me we need to make an issue to remove the `vol_idx` argument (see https://github.com/ComputationalCryoEM/ASPIRE-Python/issues/518#issuecomment-970112858).

Reply:
We have one, https://github.com/ComputationalCryoEM/ASPIRE-Python/issues/727

I had been leaving it for our junior members since it was low risk, but you are right it has been too long.

Reply:
Sorry I missed that. Maybe Josh can look at it when he's done with `OrientedSource`.

Reply:
Yeah I think we've talked about it after some recent broadcasting exercises. Its top of board for small issues in 0.11.1

Reply:
Cool.